database: Complete Prisma schema and seed script for Sprint 2

roma4in1 opened the pull request:
Database Setup - Sprint 2

Summary:
Complete database schema implementation using Prisma ORM with MySQL. Provides the foundation for all Sprint 2 features.

Database Models (7 total for now):
1. **User** - Authentication and role management (STUDENT, ORGANIZER, ADMIN)
2. **Organization** - Campus organizations hosting events
3. **Event** - Events with capacity, pricing, categories, approval workflow
4. **Ticket** - Event tickets with QR codes and payment tracking
5. **SavedEvent** - Calendar/bookmarking functionality
6. **Review** - Event ratings and comments (additional feature)

Test Data:
- 6 users (1 admin, 2 organizers, 3 students)
- 5 sample events (various categories, free and paid)
- 4 organizations (CSSA, CSU, Athletics, Engineering)
- 3 tickets (different statuses: claimed, checked-in)
- 2 saved events

Technical Changes:
- Complete Prisma schema with indexes and relationships
- Database migrations for MySQL
- seed script with data
- Package.json updated with seed script using tsx
- Environment protection (.gitignore for .env)(NEVER COMMIT .env!!!)
- server/README.md document

Setup Instructions for Reviewers:
```bash
cd server
npm install
# Create your own .env file with MySQL credentials (see server/README.md)
npx prisma generate
npx prisma migrate dev
npm run seed
npx prisma studio (Opens browser to view data)

jasdeep-s-s reviewed (commented):
So a few comments:
1. You need to provide the DATABASE_URL link that is in your .ENV file locally. It asks for that var when you run the 4th command in Setup. So share it on Discord server and NOT here! That way, I can link and copy your database to MySQL on my device. Otherwise each member will have to spin up and copy the DB locally and manually which is unnecessary and very time-consuming.
2. On that note, a friend (from another course) told me they used Azure cloud for Database access and sharing for their SOEN 341 project last sem. Everyone will be able to access the DB on the cloud and won't have to install DB locally each time. It will just be a very convenient way for DB handling plus Azure is free via Concordia email! 
3. Some changes in the .tson files are inevitable while adding DB and using Prisma ORM but do check those changes manually. I think Commonjs and ESM are both compatible with Prisma.

roma4in1 commented:
Thanks for the detailed feedback! You're absolutely right on all points. Here's what I've addressed:

1. DATABASE_URL Sharing
Good catch - I'll share the DATABASE_URL connection string on Discord (in #general and via DMs) and NOT here on GitHub for security.
2. Azure Cloud Database - Done!
implemented Azure MySQL for this reason:
- Server: `soen341-campus3.mysql.database.azure.com`
- Database: `campus_events`
- Benefit: Everyone can access the same database without local MySQL installation
- Cost: Free with Concordia email (as you mentioned!)

The setup instructions are in `server/README.md` with all the details.
 3. tsconfig.json - Fixed to ESNext
You're 100% correct about CommonJS vs ESNext! I initially changed it to `commonjs` to fix a `ts-node` issue, but that was a lazy way to do it.

Changes made:
- Reverted `tsconfig.json` to use `"module": "esnext"` (modern standard)
-  Updated dev scripts to use `tsx` instead of `ts-node` (fixes the .ts extension issue properly)
-  Updated `package.json` scripts accordingly
- Summary of Changes

I've pushed an additional commit that addresses the tsconfig concern. The PR now includes:
- Complete Prisma schema (7 models)
- Azure MySQL setup (shared database for team)
- Seed script with test data
- README documentation
- Modern ESNext TypeScript configuration
- Fixed npm scripts using `tsx`

Let me know if you'd like me to clarify anything else or make additional changes!

jasdeep-s-s reviewed (approved):
I couldn't access the DB in VS Code using the extension but it worked smoothy in MySQL workbench. I shared the schema EER diagram in the discord server and it looks good to me. You can share merge and pull to the main branch.
